### PR TITLE
Adding retries

### PR DESCRIPTION
**Pull request by alexbourret**

### CODE REVIEW

File: python-lib/api_call.py [hunk 1]
```diff
@@ -153,7 +157,19 @@ def query(url: str, headers: dict, parameters: dict) -> dict:
...
+    while not successful_get and attempt_number <= Constants.MAX_RETRIES:
+        try:
+            attempt_number += 1
+            response = requests.get(url=url, headers=headers, params=parameters)
+            successful_get = True
+        except Exception as err:
+            logger.warning("ERROR:{}".format(err))
+            logger.warning("on attempt #{}".format(attempt_number))
+            if attempt_number == Constants.MAX_RETRIES:
+                raise LinkedinPluginError("Error in batch processing on attempt #{}: {}".format(attempt_number, err))
```
Comment (Marsobad) [inline python-lib/api_call.py:171]: Maybe we could return a slightly more user-friendly message such as : "The request that you sent to LinkedIn took longer than expected despite {attempt_number} attempts. Consider decreasing the number of account ids or checking on your internet connection."

Comment (Marsobad) [reply python-lib/api_call.py:171]: Maybe, it could be nice to specify which dataset is causing the issue

Comment (alexbourret) [reply python-lib/api_call.py]: Following your comment, a suggestion to reduce the number of ids/batch size has been added in case of an error 400.

**Review by Marsobad — COMMENTED** [review]
Looks good to me !
I emulated a timeout error with timeout = 1 and your fix solves the timeout issues for ad campaigns. 
I'm just concerned that the exception messages such as timeout errors may be difficult to understand for a non technical audience. Maybe we could add fix messages such as : "retry later or filter the query" for timeout errors, and "make sure that you are connected to the internet" when it fails to establish a new connection because there's no internet.

**Review by Marsobad — COMMENTED** [review]
Looks good to me !
I emulated a timeout error with timeout = 1 and your fix solves the timeout issues for ad campaigns. 
I'm just concerned that the exception messages such as timeout errors may be difficult to understand for a non technical audience. Maybe we could add fix messages such as : "retry later or filter the query" for timeout errors, and "make sure that you are connected to the internet" when it fails to establish a new connection because there's no internet.

**Pull request merged**